tests/test_external_search.py

```diff
@@ -4120,9 +4127,14 @@ def test_unimplemented_features(self):
 
     def test_modify_search_query(self):
         class MockSearch(object):
-            called_with = "not called"
+            update_from_dict_called_with = "not called"
+            getitem_called_with = "not called"
             def update_from_dict(self, dict):
-                self.called_with = dict
+                self.update_from_dict_called_with = dict
+                return self
+
+            def __getitem__(self, slice):
+                self.getitem_called_with = slice
```

Comment:
It's now necessary to verify that our mocked-up SearchKeyPagination object gets sliced, just like a regular Pagination object would.